nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -202,24 +206,102 @@ def get_weight_tensor(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
+        :return: weight definition of the node.
+        """
+        metatype = ONNX_OPERATION_METATYPES.get_operator_metatype_by_op_name(node.op_type)
+        if metatype in WEIGHT_LAYER_METATYPES:
+            return metatype.weight_definitions
+        raise RuntimeError(f'The metatype {metatype} does not belong to a list of metatypes with a weight tensor.')
+
+    def get_weight_port_id(self, node: onnx.NodeProto) -> int:
+        """
+        Returns input port id, where a weight tensor should output.
+
+        :param node: Node, for which input port id is returned,
+        :return: input port id, where a weight tensor should output.
+        """
+        weight_definitions = self._get_weight_definitions(node)
+        if weight_definitions.weight_port_id is not None:
+            return weight_definitions.weight_port_id
+        raise RuntimeError(f'The metatype {node} does not have weight_port_id attribute')
+
+    def get_weight_channel_axis(self, node: onnx.NodeProto) -> int:
+        """
+        Returns a channel axis for weight per-channel quantization.
+
+        :param node: Node, for which weight per-channel axis id is returned,
+        :return: Channel axis for per-channel quantization.
+        """
+        weight_definitions = self._get_weight_definitions(node)
+        if weight_definitions.weight_channel_axis is not None:
+            return weight_definitions.weight_channel_axis
+        raise RuntimeError(f'The node {node} does not have weight_channel_axis attribute')
+
+    def get_bias_tensor_port_id(self, node: onnx.NodeProto) -> int:
+        """
+        Returns input port id, where a bias tensor should output.
+
+        :param node: Node, for which input port id is returned,
+        :return: input port id, where a weight bias should output.
+        """
+        weight_definitions = self._get_weight_definitions(node)
+        if weight_definitions.bias_port_id is not None:
+            return weight_definitions.bias_port_id
+        raise RuntimeError(f'The node {node} does not have bias_port_id attribute')
+
+    def _get_weight_tensor_with_reshape(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
+        """
+        Returns node's weight tensor name and its value in the case when reshape node is placed after the weight.
+        The returned weight tensor will be reshaped according to a shape attribute of the reshape node.
+
+        :param node: Reshape node, whose input is weight tensor.
+        :return: The weight tensor name and its value with applied the reshape operation.
+        """
+        tensor_name = node.output[0]
+        shape = self.get_initializers_value(node.input[1])
+        tensor_value = self.get_initializers_value(node.input[0])
+        reshaped_tensor_value = tensor_value.reshape(shape)
+        return tensor_name, reshaped_tensor_value
+
+    def _get_tensor_from_zero_input(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
```

Comment:
Do you really need this method? You use it once.

Reply:
To not overwhelm the main method I divide some logic into separate function. I don't see the problem here. Do you?